xml/src/main/java/org/ehcache/xml/ConfigurationParser.java

```diff
@@ -105,6 +112,23 @@
   private final Map<URI, CacheResourceConfigurationParser> resourceXmlParsers = new HashMap<URI, CacheResourceConfigurationParser>();
   private final ConfigType config;
 
+  static String getReplacementString(String originalValue) {
+    Matcher matcher = SYSPROP.matcher(originalValue);
+
+    StringBuffer sb = new StringBuffer();
+    while (matcher.find()) {
+      final String property = matcher.group(1);
+      final String value = System.getProperty(property);
+      if (value == null) {
+        return null;
```

Comment:
I think the simplest thing to do here is to fail... the only question that leaves is what if someone wants to include a string literal like "${blah}" in their path?  Should we support "escaping the `$`?